aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -53,18 +53,81 @@ static inline Tensor unary_op_impl(const Tensor& self, OutImpl& out_impl) {
   return out_impl(result, self);
 }
 
+// This helper function ensures the output type of the unary op is a floating point value.
+// Note that this function is only appropriate for functions like sin() which (mathematically speaking)
+// produces non-integral values. Implementers should continue to use the other unary functions, like unary_op_impl_,
+// since they explicitly specify the output type.
+template <typename OutImpl>
+static inline Tensor unary_float_op_impl(const Tensor& self, OutImpl& out_impl) {
```

Comment:
Is this still in use?

Reply:
No, sorry. I'll remove this in the next commit!

Reply:
This should be resolved here: https://github.com/pytorch/pytorch/pull/33322/commits/0a3ca09ff29927b82399d858918dca69a799c085. Thanks! :)